go/vt/topo/stats_conn_test.go

```diff
@@ -185,238 +195,216 @@ func (st *fakeConn) IsReadOnly() bool {
 }
 
 // createTestReadSemaphoreContention simulates semaphore contention on the test read semaphore.
-func createTestReadSemaphoreContention(ctx context.Context, duration time.Duration) {
+func createTestReadSemaphoreContention(ctx context.Context, duration time.Duration, semAcquiredChan chan bool) {
 	if err := testStatsConnReadSem.Acquire(ctx, 1); err != nil {
 		panic(err)
 	}
 	defer testStatsConnReadSem.Release(1)
+	semAcquiredChan <- true
 	time.Sleep(duration)
 }
 
 // TestStatsConnTopoListDir emits stats on ListDir
 func TestStatsConnTopoListDir(t *testing.T) {
+	testStatsConnStatsReset()
+	defer testStatsConnStatsReset()
+
 	conn := &fakeConn{}
 	statsConn := NewStatsConn("global", conn, testStatsConnReadSem)
 	ctx := context.Background()
 
-	go createTestReadSemaphoreContention(ctx, 100*time.Millisecond)
+	semAcquiredChan := make(chan bool)
```

Comment:
Not that it really matters here, but when you don't care about the value being passed you can use an empty struct literal which is zero bytes:
```
semAcquiredChan := make(chan struct{})

semAcquiredChan <- struct{}{}
```